#### Задача 1.

Сравните токенизатор, который вы самостоятельно писали на предыдущих занятиях, с токенизатором razdel. Какой вам больше нравится? (Возьмите для проверки какой-нибудь сложный текст - с аббревиатурами, кг/см и смайликами). 

In [ ]:
# your code here

#### Задача 2.

Возьмите известные вам сентенайзеры. Подберите достаточно сложный, по вашему мнению, текст (подсказка: можно взять какой-нибудь пост из социальной сети), протестируйте токенизацию. Какие есть различия, какой сентенайзер вам больше нравится? 

Бонус: возьмите достаточно большой текстовый файл (войну и мир, например), токенизируйте по очереди разными сентенайзерами и сравните время. Как измерить затраченное время в юпитере? Можно использовать магическую функцию %%timeit. Она возвращает среднее время, затраченное на выполнение кода в ячейке. Пример использования:

In [1]:
%%timeit
lst = [i ** 0.5 for i in range(10000)]

825 µs ± 2.95 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)


Числа здесь - количество микросекунд (одна миллионная секунды). Питон сообщает, что прогнал это выражение 7 раз, каждый раз по 1000 раз (итого - 7000 раз), на каждом выполнении замерил, сколько времени это у него заняло, посчитал среднее арифметическое (825 микросекунд) и стандартное отклонение (то есть, в среднем время выполнения команды может отличаться от среднего арифметического не больше чем на 3 микросекунды). 

In [ ]:
# your code here

#### Задача 2.5

Сравните время работы для своего токенизатора и токенизатора razdel. 

In [ ]:
# your code here

#### Задача 3. 

Возьмите любой текст на одном из языков, которые поддерживает spacy, и токенизируйте его. 

In [ ]:
# your code here